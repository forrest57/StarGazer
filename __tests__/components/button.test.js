import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'
import AnimatedButton from '../../components/animatedButton'

const prop = 'lorem'
const textComponent = () => <Text>{prop}</Text>

describe('<animatedButtton />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AnimatedButton Component={textComponent} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('receives passed text props', () => {
    const buttonComponent = renderer
      .create(
        <AnimatedButton
          Component={textComponent}
          pressFunction={() => console.log('ipsum')}
        />
      )
      .toJSON()
      const buttonText = buttonComponent.children[0].children[0].children[0]
      expect(buttonText).toBe(prop)
  })
})
