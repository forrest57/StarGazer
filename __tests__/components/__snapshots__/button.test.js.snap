// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedButtton /> renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
      "width": "auto",
    }
  }
>
  <View
    accessible={true}
    activeOpacity={1}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 6,
        "display": "flex",
      }
    }
  >
    <Text>
      lorem
    </Text>
  </View>
</View>
`;
